src/main/java/com/google/sps/Matrix.java

```diff
@@ -0,0 +1,172 @@
+enum Cell {
+    /*
+    0 --> dead
+    1 --> alive
+    2 --> alive with mask
+    3 --> alive with vaccine
+    4 --> alive with mask and vaccine
+    5 --> infected
+    6 --> infected with mask
+    7 --> infected with vaccine
+    8 --> infected with mask and vaccine
+    */
+
+    DEAD ((byte) '0'),
+    ALIVE ((byte) '1');
+  
+    private byte cellCode;
+  
+    public byte getCode() {
+        return this.cellCode;
+    }
+  
+    private Cell(byte cellCode) {
+        this.cellCode = cellCode;
+    }
+}
+
+public class Matrix {
+
+    private byte[][] matrix;
+
+    public Matrix(String stringMatrix) {
+        stringToMatrix(stringMatrix);
+    }
+
+    public void updateMatrix() {
+        int rows = this.matrix.length,
+            columns = this.matrix[0].length;
+
+
+        byte[][] tempMatrix = getMatrixCopy(this.matrix);
+
+        for (int row = 0; row < rows; row++) {
+            for (int column = 0; column < columns; column++) {
+                this.updateCell(row, column, tempMatrix);
+            }
+        }
+
+        System.out.println();
+        printMatrix(this.matrix);
+    }
+
+    private void updateCell(int row, int column, byte[][] tempMatrix) {
+        int neighbors = getNumberOfNeighbors(row, column, tempMatrix);
+
+        byte currentCellCode = tempMatrix[row][column];
+
+        if (currentCellCode == Cell.ALIVE.getCode()) {
+            if (neighbors < 2 || neighbors > 3) {
+                this.matrix[row][column] = Cell.DEAD.getCode();
+            }
+        }
+        else if (currentCellCode == Cell.DEAD.getCode()) {
+            if (neighbors == 3) {
+                this.matrix[row][column] = Cell.ALIVE.getCode();
+            }
+        }
+
+    }
+
+    private int getNumberOfNeighbors(int row, int column, byte[][] tempMatrix) {
+        int neighbors = 0;
+
+        neighbors += this.hasAliveNeighbor(row-1,column-1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row-1,column,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row-1,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column+1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row+1,column-1,tempMatrix) ? 1 : 0;
+        neighbors += this.hasAliveNeighbor(row,column-1,tempMatrix) ? 1 : 0;
+
+        return neighbors;
+    }
+
+    private boolean hasAliveNeighbor(int row, int column, byte[][] tempMatrix) {
+        int rows = tempMatrix.length,
+            columns = tempMatrix[0].length;
+
+        boolean boundaryCheck = row >= 0 && row < rows && column >= 0 && column < columns;
+        return boundaryCheck && tempMatrix[row][column] != Cell.DEAD.getCode(); 
+    }
+
+
+    private void stringToMatrix(String stringMatrix) {
+
```

Comment:
Since this string is ultimately sourced externally (from the user), you should be validating that it meets your expectations and/or handling exceptions more explicitly probably. For example, that it has commas, that rows and columns are ints, that the stringMatrix after that contains characters that can be appropriately cast to a byte w/ the values you expect.

That said, that kind of validation doesn't necessarily need to be in the prototype, but at least adding comments marking the need for it to be added would be a good idea.

Reply:
Oh yes, I agree, I was going to implement the validation, but I decided to do it until the Beta version, but you are right I need to use comments to specify my TODOs :)